fireworks/core/launchpad.py

```diff
@@ -397,6 +407,7 @@ def get_launch_by_id(self, launch_id):
         """
         m_launch = self.launches.find_one({'launch_id': launch_id})
         if m_launch:
+            m_launch = set_action_from_gridfs(m_launch, self.gridfs_fallback)
```

Comment:
Could you refactor the name:
``set_action_from_gridfs``

to the name:
``get_action_from_gridfs``

I think the code would also be clearer if there was some kind if-else outside the function. e.g., more like
```
if "gridfs_id" in m_launch.action:
    m_launch.action = get_action_from_gridfs(m_launch.action["gridfs_id"], self.gridfs_fallback)
```
This makes it clear to the developer that they don't need to worry about get_action_from_gridfs internals at all unless there is some weird gridfs thing going on and they can mentally skip over this line of code.

Reply:
Also in the example above, the function itself is not setting anything - it's just getting information

Reply:
I was seeing the things from another perspective: the `set_action_from_gridfs` function was (1) checking if the data should be retrieved from gridfs and (2) directly inserting it in the launch dictionary (thus the "set" in the name).  
If a developer has to work on the launch collection in some other point of the code, now he should be aware of the details about how the `gridfs_id` key is stored and repeat the check once more.  

I have implemented this modification following your suggestion anyway, so that we can have a better feeling of how it looks like.

Reply:
I see your point here; the new changes also result in a lot of "if statements" like this:
```
            if m_launch.get("action") and "gridfs_id" in m_launch["action"]:
```

How about something like:

```
for l in launches:
    l["action"] = get_action_from_gridfs(l["action"], self.gridfs_fallback)
```

with the ``if`` statement within the function itself.

I think the only change from your original in that case is that the getter code is simply retrieving the action rather than updating the entire launch object on your behalf. This modification just makes it more clear to me that (i) the only part of the launch that is being modified and (ii) that you are merely trying to retrieve existing information, not "setting" any new values. When I think of "setting" the action, I think that the code is somehow receiving *new* information and modifying the information accordingly (e.g., you set the action when you return a FWAction) rather than simply grabbing existing information from a database.

Reply:
Thanks for all the comments. I updated the code according to your suggestions.